pages/api/user/[id].js

```diff
@@ -0,0 +1,82 @@
+import dbConnect from "@/db/connect";
+import User from "@/db/models/user";
+import Method from "@/db/models/method";
+import Score from "@/db/models/score";
+import { getServerSession } from "next-auth";
+import Roast from "@/db/models/roast";
+
+export default async function handler(req, res) {
+  const session = await getServerSession(req, res);
+
+  if (session) {
+    const { user } = session;
+    const connection = await dbConnect();
+    const { id } = req.query;
+
+    if (!connection) {
+      return res.status(500).json({ message: "No connection to database!" });
+    }
+
+    const currentUser = await User.findOne(user)
+      .populate("roasts")
+      .populate("methods");
+
+    if (!currentUser) {
+      return res.status(401).json({ message: "You are not authorized!" });
+    }
+
+    if (req.method === "GET") {
+      const favoriteRoasts = await currentUser.roasts;
+      const pickedRoast = await favoriteRoasts.find(
+        (roast) => roast._id.toString() === id
+      );
+      const relatedMethods = await currentUser.methods.filter((relId) =>
+        relId.roastIdForMethod === id ? true : false
```

Comment:
`relId.roastIdForMethod` as name seems not to work for me as you use the `filter` method for an array with the name `methods`. Won't be `method` a better name (as a replacement for `relId`)? 🤔 Destructuring may help here too: `({roastIdForMethod}) => roastIdForMethod === id`. `? true : false` can be skipped. Should work the same then.

Reply:
Same here! I did not know you can destructure here! Really cool! Thanks